src/ADAL.PCL/AcquireTokenHandlerBase.cs

```diff
@@ -105,9 +105,15 @@ internal abstract class AcquireTokenHandlerBase
         protected UserIdentifierType UserIdentifierType { get; set; }
 
         protected bool LoadFromCache { get; set; }
-        
+
         protected bool StoreToCache { get; set; }
 
+        private const int ExpirationMarginInMinutes = 5;
+
+        private bool _tokenNearExpiry;
```

Comment:
handlers should not care id the token is near expiry. cache either returns it, or not.